### ISSUE TITLE
Implement import-rest-api for API Gateway

### ISSUE THREAD

**Issue by whummer** [localstack/localstack#171]
Example usage:

```
aws --endpoint-url=http://localhost:4567 apigateway import-rest-api --body 'Swagger.json'
```

┆Issue is synchronized with this [Jira Bug](https://localstack.atlassian.net/browse/LOC-133) by [Unito](https://www.unito.io/learn-more)

**Comment by sshogunn** [comment]
Hi everyone,

Is it going to be added any time soon? I need to configure API Gateway with Local Stack, but cannot find anything like this - https://lobster1234.github.io/2017/04/05/working-with-localstack-command-line?

**Comment by adriangalera** [comment]
Is this still in process of being implemented?
I am trying to import a swagger API definition
```
aws --endpoint-url=http://localhost:4567 apigateway import-rest-api --parameters endpointConfigurationTypes=REGIONAL --body 'file::///home/gal/Downloads/swagger.json'
```
And I can see in the logs:
```
2017-12-13T10:12:06:ERROR:localstack.services.generic_proxy: Error forwarding request: No JSON object could be decoded Traceback (most recent call last):
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 181, in forward
    path=path, data=data, headers=forward_headers)
  File "/opt/code/localstack/localstack/services/apigateway/apigateway_listener.py", line 153, in forward_request
    data = data and json.loads(to_str(data))
  File "/usr/lib/python2.7/json/__init__.py", line 339, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python2.7/json/decoder.py", line 364, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python2.7/json/decoder.py", line 382, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
```

**Comment by cfrasineanu-intx** [comment]
In my case, the swagger definition file is successfully imported (HTTP 200) but the resources/methods are not created:
`  apigatewaySetup:
    container_name: core-apigatewaySetup
    build: apigateway
    image: intx/apigateway-setup:latest
    depends_on:
      - apigateway
    environment:
      - AWS_DEFAULT_REGION=us-east-1
      - AWS_ACCESS_KEY_ID=foobar
      - AWS_SECRET_ACCESS_KEY=foobar
    entrypoint:
      - aws
      - --endpoint-url=http://10.192.1.178:4567
      - --debug
      - --no-verify-ssl
      - apigateway
      - import-rest-api
      - --body
      - 'file:///home/data/pets.json'`

`{
  "swagger": "2.0",
  "info": {
    "title": "Simple PetStore (Swagger)"
  },
  "schemes": [
    "https"
  ],
  "paths": {
    "/pets": {
      "get": {
        "responses": {
          "200": {
            "description": "200 response"
          }
        },
        "x-amazon-apigateway-integration": {
          "responses": {
            "default": {
              "statusCode": "200"
            }
          },
          "uri": "http://petstore-demo-endpoint.execute-api.com/petstore/pets",
          "passthroughBehavior": "when_no_match",
          "httpMethod": "GET",
          "type": "http"
        }
      }
    },
    "/pets/{petId}": {
      "get": {
        "parameters": [
          {
            "name": "petId",
            "in": "path",
            "required": true,
            "type": "string"
          }
        ],
        "responses": {
          "200": {
            "description": "200 response"
          }
        },
        "x-amazon-apigateway-integration": {
          "responses": {
            "default": {
              "statusCode": "200"
            }
          },
          "requestParameters": {
            "integration.request.path.id": "method.request.path.petId"
          },
          "uri": "http://petstore-demo-endpoint.execute-api.com/petstore/pets/{id}",
          "passthroughBehavior": "when_no_match",
          "httpMethod": "GET",
          "type": "http"
        }
      }
    }
  }
}`

GET http://localhost:4567/restapis/26106A-Z9592/resources returns:
`{"item": [{"path": "/", "id": "7605722565", "resourceMethods": {"GET": {}}}]}`

**Comment by steingold** [comment]
I encounter the same issue.
Has anyone managed to workaround it ?
cheers

**Comment by farazive** [comment]
+1. Same issue here.

**Comment by daviddanielarch** [comment]
+1. Same issue here, the definitions are imported correctly but nothing shows on resources.

**Comment by ChocPanda** [comment]
I am attempting to implement this in moto, I've raised this as a [feature request](https://github.com/spulec/moto/issues/2730)

**Comment by whummer** [comment]
Thanks for the update @sshogunn @adriangalera @cfrasineanu-intx @steingold @farazive @daviddanielarch-zz @ChocPanda . It's been a while since this issue was first raised, but it should now be fixed in #3762 . Please pull the `latest` Docker image and give it another try. Please let us know if there are any issues. Thanks for your help!
